src/GRA.Controllers/MissionControl/AvatarsController.cs

```diff
@@ -498,5 +493,126 @@ public async Task<IActionResult> BundleDelete(int id)
+        {
+            var layers = await _avatarService.GetLayersAsync();
+            if (layers.Any())
+            {
+                AlertDanger = "Avatars have already been set up";
+                return RedirectToAction(nameof(Index));
+            }
+
+            if (viewModel?.UploadedFile == null)
+            {
+                ShowAlertDanger("You must upload an avatar .zip file.");
+                ModelState.AddModelError(nameof(viewModel.UploadedFile),
+                    "An avatar .zip file is required.");
+                return RedirectToAction(nameof(AvatarsController.Index));
+            }
+
+            string assetPath = Path.Combine(
+                Directory.GetParent(_hostingEnvironment.WebRootPath).FullName, "assets");
+
+            if (!Directory.Exists(assetPath))
+            {
+                Directory.CreateDirectory(assetPath);
+            }
+
+            assetPath = Path.Combine(assetPath, "uploadedavatars");
+            if (Directory.Exists(assetPath))
+            {
+                Directory.Delete(assetPath, true);
+            }
+
+            Directory.CreateDirectory(assetPath);
+
+            try
+            {
+                using (var archive = new ZipArchive(viewModel.UploadedFile.OpenReadStream()))
+                {
+                    archive.ExtractToDirectory(assetPath);
+                }
+            }
+            catch (Exception ex)
+            {
+                ShowAlertDanger($"Error with avatar .zip file: {ex.Message}");
+                return RedirectToAction(nameof(AvatarsController.Index));
+            }
+
+            return await RunImportJob(assetPath);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> SetupAvatarZip()
+        {
+            var layers = await _avatarService.GetLayersAsync();
+            if (layers.Any())
+            {
+                AlertDanger = "Avatars have already been set up";
+                return RedirectToAction(nameof(Index));
+            }
+
+            var avatarZip = _pathResolver.ResolvePrivateFilePath("avatars.zip");
+
+            if (!System.IO.File.Exists(avatarZip))
+            {
+                AlertDanger = "The avatars.zip cannot be found in the shared/private folder.";
+                return RedirectToAction(nameof(Index));
+            }
+
+            string assetPath = Path.Combine(
+                Directory.GetParent(_hostingEnvironment.WebRootPath).FullName, "assets");
+
+            if (!Directory.Exists(assetPath))
+            {
+                Directory.CreateDirectory(assetPath);
+            }
+
+            assetPath = Path.Combine(assetPath, "avatarzip");
+
+            if (Directory.Exists(assetPath))
+            {
+                Directory.Delete(assetPath, true);
+            }
+
+            Directory.CreateDirectory(assetPath);
+
+            try
+            {
+                ZipFile.ExtractToDirectory(avatarZip, assetPath);
```

Comment:
Is giving an error saying "Zip entry name ends in directory separator character but contains data."

Reply:
It seems like the .NET Core ZIP capabilities don't work well with archives generated by 7-Zip. If you use the right-click -> Compress option in Windows Explorer then the archives upload okay.